### PR TITLE
Fixed ipv6 icmp

### PR DESCRIPTION
**Pull request by schoeppe**
Need ip6tables to accept ipv6-icmp not icmp.

The ip6tables command accepts the argument icmp (because it is a valid protocol in /etc/protocols), but the icmp protocol does not make sense in a ipv6 chain. You want ipv6-icmp instead!

This could lead to some hard to find bugs in dual stacked networks.

### CODE REVIEW

**Comment by hellerbarde** [comment]
is "ipv6-icmp" the same as "icmpv6"?

**Comment by schoeppe** [comment]
I am using the term ipv6-icmp because it is written like that in /etc/protocols and in iptables rules.
The official term is ICMPv6 I think - so the answer is yes.

**Comment by hellerbarde** [comment]
I just mentioned it, because I also ran into this issue, but for me to make it work, I had to enter `icmpv6` in the rule. See https://github.com/hellerbarde/ansible-role-firewall/commit/6e3bc35d810d3cd4928214972e2eea7198c64df4

**Comment by schoeppe** [comment]
I see.

As `icmpv6` is accepted by iptables as a protocol and it does not depend on /etc/protocols it may be even better to use this as the protocol name.

**Comment by hellerbarde** [comment]
> As `icmpv6` is accepted by iptables as a protocol and it does not depend on /etc/protocols it may be even better to use this as the protocol name.

I'd say so as well, it's the name mentioned in the man page (https://linux.die.net/man/8/ip6tables)

**Comment by rwojsznis** [comment]
Thanks @schoeppe for pointing that out!

I noticed that without the fix ipv6 neighbor discovery was broken on fresh hetzner node, took me a while to realize wtf is going on 😅

**Comment by hellerbarde** [comment]
@geerlingguy Can you please merge this and release it on ansible-galaxy? That would be very helpful.

File: defaults/main.yml [hunk 1]
```diff
@@ -4,9 +4,6 @@ firewall_enabled_at_boot: true
 
 firewall_allowed_tcp_ports:
   - "22"
-  - "25"
```
Comment (geerlingguy) [inline defaults/main.yml]: I'd rather limit this PR to just fixing the ip6tables ping command.

**Review by geerlingguy — CHANGES_REQUESTED** [review]

**Comment by geerlingguy** [comment]
This is in 2.4.3 — thanks again!

**Comment by hellerbarde** [comment]
Thanks a lot! *removes manual patch in company tooling* ;)

**Pull request merged**